### PR TITLE
feat: base network

### PR DESCRIPTION
**Pull request by just-toby**
## Description

Adds Base mainnet and testnet to the interface

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2387/base-mainnet-update-interface , https://linear.app/uniswap/issue/WEB-2386/base-goerli-update-interface

### Before

no base network options

### After

BASE options added to network selector:
image

BASE colors and bridge link:
(connected)
image
(unconnected)
image

### QA (ie manual testing)

- [x] manually navigated to every page in the app to ensure that the new chain doesn't cause a crash

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

**Review by JFrankfurt — CHANGES_REQUESTED** [review]
I don't think we want this rpc to be the one people add to their wallets. Look for the "safe rpc" lists.

image

File: src/components/NavBar/SearchBarDropdown.tsx [hunk 1]
```diff
@@ -362,6 +362,9 @@ function ComingSoonText({ chainId }: { chainId: ChainId }) {
       return <Trans>Coming soon: search and explore tokens on BNB Chain</Trans>
     case ChainId.AVALANCHE:
       return <Trans>Coming soon: search and explore tokens on Avalanche Chain</Trans>
+    case ChainId.BASE:
+    case ChainId.BASE_GOERLI:
```
Comment (cbachmeier) [inline src/components/NavBar/SearchBarDropdown.tsx:366]: I'm weirdly not seeing this on Base_
Screenshot 2023-07-26 at 3 21 07 PM
Goerli

File: src/utils/getExplorerLink.test.ts [hunk 2]
```diff
@@ -35,4 +35,12 @@ describe('#getExplorerLink', () => {
       'https://snowtrace.io/address/abc'
     )
   })
+  it('base', () => {
+    expect(getExplorerLink(ChainId.BASE, 'abc', ExplorerDataType.ADDRESS)).toEqual('https://basescan.io/address/abc')
```
Comment (cbachmeier) [inline src/utils/getExplorerLink.test.ts:39]: `basescan.org`

File: src/utils/getExplorerLink.test.ts [hunk 3]
```diff
@@ -35,4 +35,12 @@ describe('#getExplorerLink', () => {
...
+  })
+  it('base goerli', () => {
+    expect(getExplorerLink(ChainId.BASE_GOERLI, 'abc', ExplorerDataType.ADDRESS)).toEqual(
+      'https://goerli.basescan.io/address/abc'
```
Comment (cbachmeier) [inline src/utils/getExplorerLink.test.ts:43]: `basescan.org`

File: src/utils/getExplorerLink.ts [hunk 4]
```diff
@@ -12,6 +12,8 @@ const BLOCK_EXPLORER_PREFIXES: { [chainId: number]: string } = {
   [ChainId.CELO_ALFAJORES]: 'https://alfajores-blockscout.celo-testnet.org',
   [ChainId.BNB]: 'https://bscscan.com',
   [ChainId.AVALANCHE]: 'https://snowtrace.io',
+  [ChainId.BASE]: 'https://basescan.io',
```
Comment (cbachmeier) [inline src/utils/getExplorerLink.ts:15]: `basescan.org`

File: src/utils/getExplorerLink.ts [hunk 5]
```diff
@@ -12,6 +12,8 @@ const BLOCK_EXPLORER_PREFIXES: { [chainId: number]: string } = {
...
+  [ChainId.BASE_GOERLI]: 'https://goerli.basescan.io',
```
Comment (cbachmeier) [inline src/utils/getExplorerLink.ts:16]: `basescan.org`

File: src/components/NetworkAlert/NetworkAlert.tsx [hunk 6]
```diff
@@ -150,6 +160,8 @@ const TEXT_COLORS: { [chainId in NetworkAlertChains]: string } = {
   [ChainId.BNB]: colors.gold400,
   [ChainId.ARBITRUM_GOERLI]: '#0490ed',
   [ChainId.AVALANCHE]: '#ff3856',
+  [ChainId.BASE]: '#0052FF',
+  [ChainId.BASE_GOERLI]: '#0052FF',
```
Comment (cbachmeier) [inline src/components/NetworkAlert/NetworkAlert.tsx:164]: `colors.networkBase`

File: src/constants/chainInfo.ts [hunk 7]
```diff
@@ -241,6 +243,38 @@ const CHAIN_INFO: ChainInfoMap = {
     color: darkTheme.chain_43114,
     backgroundColor: darkTheme.chain_43114_background,
   },
+  [ChainId.BASE]: {
+    networkType: NetworkType.L2,
+    blockWaitMsBeforeWarning: ms`25m`,
+    bridge: 'https://bridge.base.org/deposit',
```
Comment (cbachmeier) [inline src/constants/chainInfo.ts:249]: this is probably more for the Base team but this is auto-routing to goerli currently, when they go live this will be the right url?

File: src/constants/lists.ts [hunk 8]
```diff
@@ -24,6 +24,8 @@ export const CELO_LIST = 'https://celo-org.github.io/celo-token-list/celo.tokenl
 export const PLASMA_BNB_LIST = 'https://raw.githubusercontent.com/plasmadlt/plasma-finance-token-list/master/bnb.json'
 export const AVALANCHE_LIST =
   'https://raw.githubusercontent.com/ava-labs/avalanche-bridge-resources/main/token_list.json'
+export const BASE_LIST =
+  'https://raw.githubusercontent.com/ethereum-optimism/ethereum-optimism.github.io/master/optimism.tokenlist.json'
```
Comment (cbachmeier) [inline src/constants/lists.ts:28]: optimism? Is this a placeholder?

File: src/constants/networks.ts [hunk 9]
```diff
@@ -143,8 +151,20 @@ export const RPC_URLS = {
...
-  [ChainId.BNB]: [QUICKNODE_RPC_URL, ...FALLBACK_URLS[ChainId.BNB]],
+  [ChainId.BNB]: [QUICKNODE_BNB_RPC_URL, ...FALLBACK_URLS[ChainId.BNB]],
   [ChainId.AVALANCHE]: [`https://avalanche-mainnet.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.AVALANCHE]],
-  [ChainId.BASE]: [`https://base-mainnet.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.BASE]],
-  [ChainId.BASE_GOERLI]: [`https://base-goerli.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.BASE_GOERLI]],
+  [ChainId.BASE]: [
+    `https://base-mainnet.infura.io/v3/${INFURA_KEY}`,
+    QUICKNODE_BASE_RPC_URL,
+    'https://base-mainnet.blastapi.io/b5a802d8-151d-4443-90a7-699108dc4e01',
+    'https://svc.blockdaemon.com/base/mainnet/native?apiKey=zpka_1334e7c450464d06b6e33a972a7a4e57_75320f43',
```
Comment (cbachmeier) [inline src/constants/networks.ts:160]: add these to Fallback URLS under an `Unsafe` commetn

File: src/constants/networks.ts [hunk 10]
```diff
@@ -143,8 +151,20 @@ export const RPC_URLS = {
...
+    ...FALLBACK_URLS[ChainId.BASE],
+  ],
+  [ChainId.BASE_GOERLI]: [
+    `https://base-goerli.infura.io/v3/${INFURA_KEY}`,
+    QUICKNODE_BASE_GOERLI_RPC_URL,
+    'https://base-goerli.blastapi.io/b5a802d8-151d-4443-90a7-699108dc4e01',
+    'https://svc.blockdaemon.com/base/testnet/native?apiKey=zpka_1334e7c450464d06b6e33a972a7a4e57_75320f43',
```
Comment (cbachmeier) [inline src/constants/networks.ts:167]: add these to Fallback URLS under an `Unsafe` commetn

File: src/constants/routing.ts [hunk 11]
```diff
@@ -161,6 +161,8 @@ export const COMMON_BASES: ChainCurrencyList = {
   ],
   [ChainId.OPTIMISM]: [nativeOnChain(ChainId.OPTIMISM), OP, DAI_OPTIMISM, USDC_OPTIMISM, USDT_OPTIMISM, WBTC_OPTIMISM],
   [ChainId.OPTIMISM_GOERLI]: [nativeOnChain(ChainId.OPTIMISM_GOERLI)],
+  [ChainId.BASE]: [nativeOnChain(ChainId.BASE), WRAPPED_NATIVE_CURRENCY[ChainId.BASE] as Token],
+  [ChainId.BASE_GOERLI]: [nativeOnChain(ChainId.BASE_GOERLI), WRAPPED_NATIVE_CURRENCY[ChainId.BASE_GOERLI] as Token],
```
Comment (cbachmeier) [inline src/constants/routing.ts:165]: I'm torn on this. They're the same address but this might be confusing to a dev picking this up if they didn't know Base was built on OP. I'd either add the duplicate `WRAPPED_NATIVE_CURRENCY` entry for Base and Base_Goerli, or add comments. Preferably the former.

File: src/constants/lists.ts (hunk 8 above)
Comment (just-toby) [reply src/constants/lists.ts:28]: actually no, they use the optimism "superchain" list

File: src/constants/routing.ts (hunk 11 above)
Comment (just-toby) [reply src/constants/routing.ts:165]: i'm confused by this comment - `WRAPPED_NATIVE_CURRENCY` has entries for the base networks

File: src/constants/networks.ts [hunk 12]
```diff
@@ -143,8 +159,8 @@ export const RPC_URLS = {
   ],
   [ChainId.CELO]: FALLBACK_URLS[ChainId.CELO],
   [ChainId.CELO_ALFAJORES]: FALLBACK_URLS[ChainId.CELO_ALFAJORES],
-  [ChainId.BNB]: [QUICKNODE_RPC_URL, ...FALLBACK_URLS[ChainId.BNB]],
+  [ChainId.BNB]: [QUICKNODE_BNB_RPC_URL, ...FALLBACK_URLS[ChainId.BNB]],
   [ChainId.AVALANCHE]: [`https://avalanche-mainnet.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.AVALANCHE]],
-  [ChainId.BASE]: [`https://base-mainnet.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.BASE]],
-  [ChainId.BASE_GOERLI]: [`https://base-goerli.infura.io/v3/${INFURA_KEY}`, ...FALLBACK_URLS[ChainId.BASE_GOERLI]],
+  [ChainId.BASE]: [`https://base-mainnet.infura.io/v3/${INFURA_KEY}`],
+  [ChainId.BASE_GOERLI]: [`https://base-goerli.infura.io/v3/${INFURA_KEY}`],
```
Comment (cbachmeier) [inline src/constants/networks.ts:165]: why are we removing the fallback urls?

File: src/components/Web3Status/index.tsx [hunk 13]
```diff
@@ -115,11 +114,6 @@ const Text = styled.p`
   font-weight: 500;
 `
 
-// we want the latest one to come first, so return negative if a is after b
-function newTransactionsFirst(a: TransactionDetails, b: TransactionDetails) {
-  return b.addedTime - a.addedTime
-}
-
```
Comment (cbachmeier) [inline src/components/Web3Status/index.tsx:122]: nit: this seems unrelated to base, it'd be better to have the linter fixes in their own PR instead of adding them to this one

File: src/components/AccountDrawer/index.tsx [hunk 14]
```diff
@@ -213,14 +213,14 @@ function AccountDrawer() {
...
-    onDragEnd: (state) => {
+    onDragEnd: () => {
       setYPosition(0)
       if (scrollRef.current) {
         scrollRef.current.style.overflowY = 'auto'
       }
     },
     // set dragStartTop to true if the user starts dragging from the top of the drawer
-    onDragStart: (state) => {
+    onDragStart: () => {
```
Comment (cbachmeier) [inline src/components/AccountDrawer/index.tsx:223]: nit: this seems unrelated to base, it'd be better to have the linter fixes in their own PR instead of adding them to this one

File: src/components/NavBar/SearchBarDropdown.tsx (hunk 1 above)
Comment (just-toby) [reply src/components/NavBar/SearchBarDropdown.tsx:366]: closing the loop on this - it's working for me!

**Review by JFrankfurt — APPROVED** [review]

**Review by cbachmeier — APPROVED** [review]
